---
title: Metabolomic characterization of neurodevelopmental disorders with neurotransition
  components
author: "Sofia Illescas"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_knit$set(root.dir = 'C:/Users/killescas/Documents/GitHub/metabolomics/LCR/metabolomica')
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)
```

```{r include=FALSE}
library(purrr)
library(imputeTS)
library(reshape2)
library(ggpubr)
library(rstatix)
library(ggplot2)
library(ggbiplot)
library(ropls)
library(knitr)
library(factoextra)
library(ggrepel)
library(janitor)
library(patchwork)
library(ggplotify)
library(here)
library(tidyverse)
```



```{r fig_1}
df_norm_lab <- readRDS(here("data_processed/wide_format_df.rds"))
log_df <- readRDS(here("data_processed/long_format_df.rds"))

metdat <- read.csv(here("metadata/metdat.csv"), header=TRUE,row.names = 1,sep=";")



```



```{r}
lcp <- c("#95C65C","#808B96","#F19743", "#C70E7B","#0ab2fa")
names(lcp) <- c("1","2","8","10","12")
lcps <- lcp[c("1","2","8","10","12")]
df_norm_lab_s <- df_norm_lab
pca_m1 <- prcomp(df_norm_lab_s[,-1], scale. = TRUE, center = TRUE)

# Get the maximum x-axis limit among all the plots
max_x_limit <- max(abs(pca_m1$x[, 1]))+2

# Create a list to store the modified plots
modified_plots <- list()

p1 <- ggbiplot(pca_m1,
                choices = c(1,2),
                obs.scale = 1,
                var.scale = 1,
                var.axes = FALSE,
                circle = TRUE,
                ellipse = FALSE,
                groups = df_norm_lab_s$Label,
                fill = TRUE) +
    theme_light(base_size = 10) +
    labs(colour = "Mutated gene") +
    geom_point(aes(color = as.factor(df_norm_lab_s$Label)), size = 4) +
    scale_color_manual(values = lcps, labels = c("Control", "MeCP2","STXBP1","CDKL5","GRIN")) +
    xlim(-max_x_limit, max_x_limit)  +
  ylim(-6,6) 


p1 <- wrap_elements(as.grob(p1))

```

```{r}
names(lcp) <- c("1","2","8","10","12")
lcps <- lcp[c("1","2","10")]
df_norm_lab_s <- subset(df_norm_lab, Label%in%c("2","10","1"))
pca_m2 <- prcomp(df_norm_lab_s[,-1], scale. = TRUE, center = TRUE)

# Get the maximum x-axis limit among all the plots
max_x_limit <- max(abs(pca_m2$x[, 1]))+2

# Create a list to store the modified plots
modified_plots <- list()

p2 <- ggbiplot(pca_m2,
                choices = c(1,2),
                obs.scale = 1,
                var.scale = 1,
                var.axes = FALSE,
                circle = TRUE,
                ellipse = FALSE,
                groups = as.factor(df_norm_lab_s$Label),
                fill = TRUE) +
    theme_light(base_size = 10) +
    labs(colour = "Mutated gene") +
    geom_point(aes(color = as.factor(df_norm_lab_s$Label)), size = 4) +
    scale_color_manual(values = lcps, labels = NULL) +
    xlim(-max_x_limit, max_x_limit) +
  ylim(-6,6) +
    theme(legend.position="none") 

p2 <- wrap_elements(as.grob(p2))

```


```{r}
df_norm_lab_s <- subset(df_norm_lab, Label%in%c("12","8","1"))
pca_m3 <- prcomp(df_norm_lab_s[,-1], scale. = TRUE, center = TRUE)
lcps <- lcp[c("1","8","12")]
# Get the maximum x-axis limit among all the plots
max_x_limit <- max(abs(pca_m3$x[, 1]))+2

# Create a list to store the modified plots
modified_plots <- list()

p3 <- ggbiplot(pca_m3,
                choices = c(1,2),
                obs.scale = 1,
                var.scale = 1,
                var.axes = FALSE,
                circle = TRUE,
                ellipse = FALSE,
                groups = as.factor(df_norm_lab_s$Label),
                fill = TRUE) +
    theme_light(base_size = 10) +
    labs(colour = "Mutated gene") +
    geom_point(aes(color = as.factor(df_norm_lab_s$Label)), size = 4) +
    scale_color_manual(values = lcps, labels = c("Control","STXBP1","GRIN")) +
    xlim(-max_x_limit, max_x_limit)  +
    ylim(-6,6) +
  theme(legend.position="none")

p3 <- wrap_elements(as.grob(p3))
```


```{r}
library(ggbiplot)
df_norm_lab4 <- df_norm_lab
df_norm_lab4$Label <- ifelse(df_norm_lab4$Label==2,10,df_norm_lab4$Label)
df_norm_lab4$Label <- ifelse(df_norm_lab4$Label==12,8,df_norm_lab4$Label)

lcp3 <- c("#95C65C","#B25D91","#FF3F38")
pca_m <- prcomp(df_norm_lab4[,-1], scale. = TRUE, center = TRUE)

# Get the maximum x-axis limit among all the plots
max_x_limit <- max(abs(pca_m$x[, 1]))+2

df_norm_lab_s <- df_norm_lab4
p4 <- ggbiplot(pca_m,
                choices = c(1,2),
                obs.scale = 1,
                var.scale = 1,
                var.axes = FALSE,
                circle = TRUE,
                ellipse = FALSE,
                groups = as.factor(df_norm_lab_s$Label),
                fill = TRUE) +
    theme_light(base_size = 10) +
    labs(colour = "Mutated gene") +
    geom_point(aes(color = as.factor(df_norm_lab_s$Label)), size = 4) +
    scale_color_manual(values = lcp3, labels = c("Control","Hyperglutamatergic","Hypoglutamatergic")) +
    xlim(-max_x_limit, max_x_limit)  +
    ylim(-6,6) +
  theme(legend.position="bottom")

p4 <- wrap_elements(as.grob(p4))

# Create a list to store the modified plots

# library(imager)
# neurons <- load.image("/Users/sofiaillescas/Desktop/LAB/LCR/metabolomica/neuronas.jpg")
# 
# nr <- ggplot() +
#   background_image(neurons) 

```


```{r}
df_meta <- df_norm_lab
# df_metg <- subset(df_metg,Label%in%c(1,10,2,12,8))
df_meta <- data.frame(t(df_meta),check.names = FALSE)
mat_alla <- as.matrix(df_meta)
laba <- data.frame(as.factor(mat_alla[1,]))
laba$as.factor.mat_alla.1.... <- gsub(12,"STXBP1  ",laba$as.factor.mat_alla.1....)
laba$as.factor.mat_alla.1.... <- gsub(2,"CDKL5  ",laba$as.factor.mat_alla.1....)
laba$as.factor.mat_alla.1.... <- gsub(10,"MeCP2  ",laba$as.factor.mat_alla.1....)
laba$as.factor.mat_alla.1.... <- gsub(1,"Control  ",laba$as.factor.mat_alla.1....)
laba$as.factor.mat_alla.1.... <- gsub(8,"GRIN  ",laba$as.factor.mat_alla.1....)
laba$as.factor.mat_alla.1.... <- gsub("STXBPControl    ","STXBP1  ",laba$as.factor.mat_alla.1....)
colnames(laba)<-"Label"
laba$Label<- factor(laba$Label,levels = c("Control  ","MeCP2  ","CDKL5  ", "STXBP1  ", "GRIN  "),ordered = TRUE)

mat_alla <- apply(mat_alla[-1,],c(1,2),as.numeric)

library(viridis)
library(pheatmap)

lc <- c("#95C65C","#C70E7B","#808B96","#46dbf2","#F19743")
names(lc) <-  levels(laba$Label)
lc <- list(Label = lc)


pheatmap(mat_alla, annotation = laba,clustering_method = "ward.D2",fontsize = 8,show_colnames = TRUE,show_rownames = FALSE,main = "Comparing patients to controls", border_color = NA, annotation_colors = lc, fontsize_col = 10, color = colorRampPalette(c("blue","red"))(100), cluster_rows = TRUE,scale = "row")
```

```{r fig.width = 5, fig.height = 4, fig.show = "hold", out.height="50%",out.width="50%",  results='asis'}
df_metr <- df_norm_lab
df_metr <- subset(df_metr,Label%in%c(1,10,2))
rownames(df_metr) <- subset(metdat,metdat$COS.Code%in%rownames(df_metr))$Gene.ID
df_metr <- data.frame(t(df_metr),check.names = FALSE)
mat_allr <- as.matrix(df_metr)
conds <- c("Hypoglutamatergic","Control")
labr <- data.frame(as.factor(mat_allr[1,]))

# rownames(labr) <- m
labr$as.factor.mat_allr.1.... <- gsub(2,"CDKL5  ",labr$as.factor.mat_allr.1....)
labr$as.factor.mat_allr.1.... <- gsub(10,"MeCP2  ",labr$as.factor.mat_allr.1....)
labr$as.factor.mat_allr.1.... <- gsub(1,"Control  ",labr$as.factor.mat_allr.1....)
colnames(labr)<-"Label"
labr$Label<- factor(labr$Label,levels = c("Control  ","MeCP2  ","CDKL5  "),ordered = TRUE)

mat_allr <- apply(mat_allr[-1,],c(1,2),as.numeric)

library(viridis)
library(pheatmap)

lc <- c("#95C65C","#C70E7B","#808B96")
names(lc) <-  levels(labr$Label)
lc <- list(Label = lc)

# colnames(mat_allr) <- subset(metdat,metdat$COS.Code%in%colnames(mat_allr))$Gene.ID


pheatmap(mat_allr, annotation = labr,clustering_method = "ward.D2",fontsize = 8,show_colnames = TRUE,show_rownames = FALSE, border_color = NA, annotation_colors = lc, fontsize_col = 10, color = colorRampPalette(c("blue","red"))(100), cluster_rows = TRUE,scale = "row",annotation_legend = TRUE)

```


```{r fig.width = 5, fig.height = 4}
df_metg <- df_norm_lab 
df_metg <- subset(df_metg,Label%in%c(1,8,12))
rownames(df_metg) <- subset(metdat,metdat$COS.Code%in%rownames(df_metg))$Gene.ID
df_metg <- data.frame(t(df_metg),check.names = FALSE)
mat_allg <- as.matrix(df_metg)
conds <- c("GRIN","Control")
lab <- data.frame(as.factor(mat_allg[1,]))
lab$as.factor.mat_allg.1.... <- gsub(8,"GRIN",lab$as.factor.mat_allg.1....)
lab$as.factor.mat_allg.1.... <- gsub(12,"STXBP1",lab$as.factor.mat_allg.1....)
lab$as.factor.mat_allg.1.... <- gsub(1,"Control",lab$as.factor.mat_allg.1....)
lab$as.factor.mat_allg.1.... <- gsub("STXBPControl","STXBP1  ",lab$as.factor.mat_allg.1....)
colnames(lab)<-"Label"
lab$Label<- factor(lab$Label,levels = c("Control","STXBP1  ","GRIN"),ordered = TRUE)

mat_allg <- apply(mat_allg[-1,],c(1,2),as.numeric)

lc <- c("#95C65C","#46dbf2","#F19743")
names(lc) <-  levels(lab$Label)
lc <- list(Label = lc)

p5 <- pheatmap(mat_allg, annotation = lab,clustering_method = "ward.D2",fontsize = 8,show_colnames = TRUE,show_rownames = FALSE, border_color = NA, annotation_colors = lc, fontsize_col = 10, color = colorRampPalette(c("blue","red"))(100), cluster_rows = TRUE,scale = "row",legend = TRUE, annotation_legend = TRUE)

p5 <- wrap_elements(as.grob(p5))


# (nr|(p1/(p2+p3)/(p4+p5))+plot_layout(heights = c(1,2,2)))+plot_layout(guides = "collect")+plot_annotation(tag_levels = 'A') 
```

# Study design

## Univariate Analysis

```{r table_3, results = 'asis'}
lcp3 <- c("#FF3F38","#B25D91","#95C65C")
df_norm_lab4 <- df_norm_lab
df_norm_lab4$Label <- ifelse(df_norm_lab4$Label==2,10,df_norm_lab4$Label)
df_norm_lab4$Label <- ifelse(df_norm_lab4$Label==12,8,df_norm_lab4$Label)

df_norm_lab4 <- rownames_to_column(df_norm_lab4)
log_df1 <- melt(df_norm_lab4, id.vars=c("rowname","Label"))
ex_df1 <- log_df1
ex_df1$value <- exp(ex_df1$value)
ex_df1$value <-as.numeric(format(ex_df1$value, scientific = FALSE))
ex_df1$Label <- factor(ex_df1$Label)
# 
# sam_stat <- log_df %>%
#   group_by(Label,variable) %>%
#   get_summary_stats(value, type = "median_iqr")

stat.test <-
    log_df1 %>%
    group_by(variable) %>%
    wilcox_test(value ~ Label) %>%
    adjust_pvalue(method="fdr") %>%
    add_significance("p.adj")


sta_lst_sig <- stat.test
 

```


## Multivariate Analysis

```{r }

pd_g <- list()
vip_g <- list()
var_g <- list()



for (j in c(10,8)) {
  df_norm_lab5 <- subset(df_norm_lab4, Label%in%c(j,1))
  rownames(df_norm_lab5) <- df_norm_lab5$rowname
  pd_g <- append(pd_g,opls(df_norm_lab5[,-c(1,2)],as.factor(df_norm_lab5$Label),orthoI = 1,predI=1,scaleC = "center", permI = 1000))
  vip_g <- lapply(pd_g, function(x) data.frame(getVipVn(x)))
  vip_g <- lapply(vip_g, function(x) {colnames(x) <- "VIP"; x})
  vip_g <- lapply(vip_g, function(x) {x$Metabolite <- rownames(x); x})
  var_g <- lapply(vip_g, function(x) data.frame(x$Metabolite[which(x$VIP>1)]))
}

names(pd_g) <- c("Hyperglutamatergic","Hypoglutamatergic")
tables <- lapply(pd_g, function(x) getSummaryDF(x))
lapply(names(tables),function(x) write.csv(tables[[x]],file=paste0("opls_diag_",x,".csv")))


names(var_g) <- c("Hyperglutamatergic","Hypoglutamatergic")
names(vip_g) <- c("Hyperglutamatergic","Hypoglutamatergic")



col_op <- c("#B25D91", "#FF3F38")
names(col_op) <- names(vip_g)

sels <- c()  # Create an empty list to store plots

for (i in 1:length(vip_g)) {
  vip_g2 <- vip_g[[i]][order(vip_g[[i]]$VIP, decreasing = TRUE),]
  vip_g2 <- subset(vip_g2, VIP > 1)
  vip_g2$color <- names(vip_g[i])
  sels <- append(sels, vip_g2$Metabolite)
  
  p <- vip_g2 %>%
    filter(!is.na(VIP)) %>%
    arrange(VIP) %>%
    mutate(Metabolite = factor(Metabolite, Metabolite)) %>%
    ggplot() +
    geom_point(aes(x = Metabolite, y = VIP, size = 3, color = color)) +
    coord_flip() +
    theme(
      panel.grid.minor.y = element_blank(),
      panel.grid.major.y = element_blank(),
      legend.position = "right"
    ) +
    theme_light() +
    xlab("") +
    ylab("VIP scores") +
    ggtitle(c("Hyperglutamatergic", "Hypoglutamatergic")[i]) +
    ylim(0, 3.5) +
    guides(size = "none") +
    scale_color_manual(values = col_op[i], labels=NULL)
  
  plot(p) 
}

vip_g$Hypoglutamatergic$p.adj <- subset(sta_lst_sig,group2=="8"&group1=="1")$p.adj
vip_g$Hyperglutamatergic$p.adj <- subset(sta_lst_sig,group2=="10"&group1=="1")$p.adj

```




```{r}
library(VennDiagram)

col_op <- c("#B25D91", "#FF3F38")
names(col_op) <- names(vip_g)

sels <- c()  # Create an empty list to store plots

for (i in names(vip_g)) {
  vip_g2 <- vip_g[[i]][order(vip_g[[i]]$VIP, decreasing = TRUE),]
  vip_g2 <- subset(vip_g2, VIP > 1 | p.adj <=0.05)
  vip_g2$color <- names(vip_g[i])
  sels[[i]] <- vip_g2$Metabolite
}

venn.diagram(sels,filename = here("paper/Figures/venn.png"),
             imagetype = "png",
          col=c("#B25D91", "#FF3F38"),
          fill = c(alpha("#B25D91",0.3), alpha('#FF3F38',0.3)),
          cex = 1,
          fontfamily = "sans",
          cat.cex = 1,
          cat.default.pos = "outer",
          cat.pos = c(-27, 27),
          cat.dist = c(0.055, 0.055),
          cat.fontfamily = "sans",
          cat.col = c("#B25D91", "#FF3F38"))

cat(setdiff(sels$Hyperglutamatergic,sels$Hypoglutamatergic),sep = "\n")
cat(setdiff(sels$Hypoglutamatergic,sels$Hyperglutamatergic),sep = "\n")
cat(intersect(sels$Hyperglutamatergic,sels$Hypoglutamatergic),sep = "\n")
```

# Affected pathways

```{r, results='asis'}
# library(httr)
# library(KEGGREST)
# 
# 
# background <- read.csv(here("metadata/background.csv"))
# rows_to_remove <- grep("carnitine", background$Query)
# background <- background[-rows_to_remove, ]
# background$Query <- colnames(df_norm_lab[,-1])
# 
# # List of compound IDs
# compound_ids <- na.omit(background$KEGG)
# 
# # Base URL for KEGG API
# base_url <- "http://rest.kegg.jp"
# response <- GET("http://rest.kegg.jp/link/pathway/C00327")
# # Function to retrieve pathways for a compound
# get_pathways <- function(compound_id) {
#   request_url <- paste0(base_url, "/link/pathway/", compound_id)
#   response <- GET(request_url)
#   if (status_code(response) == 200) {
#     pathway_lines <- strsplit(content(response, "text"), "\n")[[1]]
#     pathways <- sapply(strsplit(pathway_lines, "\t"), `[`, 2)
#     return(pathways)
#   } else {
#     return(NULL)
#   }
# }
# print(content(response, "text"))
# 
# # Retrieve pathways for each compound
# all_pathways <- lapply(compound_ids, get_pathways)
# all_pathways <- lapply(all_pathways,function(x) gsub("path:","",x))
# names(all_pathways) <- compound_ids
# freq <- table(unlist(all_pathways))
# 
# # Flatten the list of pathways
# pathways <- na.omit(unique(unlist(all_pathways)))
# 
# 
# path_names <- lapply(pathways, function(x) keggFind("pathway",x))
# names(path_names) <- pathways
# 
# freq <- freq[pathways]
# names(freq) <- path_names
# freq <- data.frame(freq)
# colnames(freq) <- c("Pathway","Total Hits")
# 
# 
#   gen_paths <- c("Metabolic pathways",
#   "Biosynthesis of secondary metabolites",
#   "Microbial metabolism in diverse environments",
#   "Carbon metabolism",
#   "2-Oxocarboxylic acid metabolism",
#   "Fatty acid metabolism",
#   "Biosynthesis of amino acids",
#   "Nucleotide metabolism",
#   "Biosynthesis of nucleotide sugars",
#   "Biosynthesis of cofactors",
#   "Degradation of aromatic compounds")
# 
#   cancer_paths <- c( "Pathways in cancer", "Transcriptional misregulation in cancer"," MicroRNAs in cancer", "Proteoglycans in cancer", "Chemical carcinogenesis - DNA adducts", "Chemical carcinogenesis - receptor activation", "Chemical carcinogenesis - reactive oxygen species", "Viral carcinogenesis", "Central carbon metabolism in cancer", "Choline metabolism in cancer", "PD-L1 expression and PD-1 checkpoint pathway in cancer")
# 
#   dig_sys <- c("Salivary secretion","Gastric acid secretion","Pancreatic secretion","Bile secretion","Carbohydrate digestion and absorption","Protein digestion and absorption","Fat digestion and absorption", "Cholesterol metabolism","Vitamin digestion and absorption","Mineral absorption")
# 
#   exc_sys <- c("Vasopressin-regulated water reabsorption","Aldosterone-regulated sodium reabsorption","Endocrine and other factor-regulated calcium reabsorption","Proximal tubule bicarbonate reclamation","Collecting duct acid secretion")
# 
# p <- list()
# # s<-sels$Hyperglutamatergic
# for (s in names(sels)) {
#   sels_kegg <- subset(background, Query%in%sels[[s]])
#   names_sel <- sels_kegg$Query
#   sels_kegg <- sels_kegg$KEGG #now we need to select these metabolites in all_pathways to make the graph we want
#   names(sels_kegg) <- names_sel
# 
#   paths_sel <- all_pathways[na.omit(sels_kegg)]
#   names(paths_sel) <- names(na.omit(sels_kegg))
# 
#   paths_list <- lapply(paths_sel, function(x) data.frame(path_names[x]))
# 
#   for (i in seq_along(paths_list)) {
#     rownames(paths_list[[i]]) <- names(paths_list)[[i]]
#   }
# 
#   for (i in seq_along(paths_list)) {
#     colnames(paths_list[[i]]) <- paths_list[[i]][1,]
#   }
# 
#   paths_df <- bind_rows(paths_list)
#   paths_df <- data.frame(ifelse(is.na(paths_df),0,1), check.names = FALSE)
#   sel_sreq <- data.frame(colSums(paths_df))
#   colnames(sel_sreq) <- "Hits"
#   sel_sreq <- subset(sel_sreq, Hits>1)
#   sel_sreq <- sel_sreq %>%
#     arrange(-Hits)
# 
# 
#   sel_sreq <- subset(sel_sreq, !(rownames(sel_sreq) %in% c(gen_paths, cancer_paths, dig_sys, exc_sys)))
# 
# 
#   human <- keggList("pathway", "hsa")
# 
#   human <- sapply(human, function(x) gsub(" - Homo sapiens \\(human)","",x))
#   sel_sreq <- subset(sel_sreq, rownames(sel_sreq)%in%human)
# 
# 
#   all_freq <- subset(freq, Pathway%in%rownames(sel_sreq))
# 
# 
#   proptn <- data.frame(sel_sreq[order(as.character(rownames(sel_sreq))),]/all_freq[order(as.character(all_freq$Pathway)),]$`Total Hits`)
#   colnames(proptn) <- "Proportion"
#   proptn$Pathway <- all_freq[order(as.character(all_freq$Pathway)),]$Pathway
#   rownames(proptn) <-   proptn$Pathway
#   sel_sreq$Proportion <- proptn[rownames(sel_sreq),]$Proportion
#   sel_sreq$Pathway <- rownames(sel_sreq)
# 
# 
#   g <- sel_sreq %>%
#       arrange(Hits) %>%
#       mutate(Pathway = factor(Pathway, Pathway)) %>%
#       ggplot(aes(x = Pathway, y = Hits, fill=Proportion)) +
#       ylim(0,7) +
#       geom_col(size=2) +
#       coord_flip() +
#       theme(
#         panel.grid.minor.y = element_blank(),
#         panel.grid.major.y = element_blank(),
#         legend.position = "right",
#         legend.spacing = unit(1, 'cm')
#       ) +
#       theme_light(base_size = 15) +
#       xlab("") +
#       ylab("Hits") +
#       scale_fill_gradient2(low="#F4E3C7",mid = "#ff7f75",high =  "#aa1a00", limits=c(0,1), midpoint = 0.4) +
#       guides(size = "none")
# 
#   p[[s]] <- g
# 
# }
# 
# 
# p$Hypoglutamatergic <- p$Hypoglutamatergic + scale_x_discrete(position = "top")
# 
# patchpath <- p$Hyperglutamatergic + p$Hypoglutamatergic + plot_annotation("Pathways affected by selected metabolites") +plot_layout(guides = "collect") & theme(legend.position = 'right',legend.spacing = unit(1, 'cm'))
# 
# # 
# # ggsave(file=here("paper/Figures/paths.jpeg"),plot=patchpath,       width = 300, 
# #        height = 200, 
# #        units = "mm", dpi = 300, scale = 0.8) 
# 
# ggsave(file=here("paper/Figures/paths.png"),patchpath,dpi = 1000, scale = 1,width = 40, height = 15, units = "cm")
```

# Alterations in tryptophan metabolism

```{r}


tryp_mets <- c(
"Kynurenic acid",
"Kynurenine",
"3-Hydroxyanthranilic acid",
"Tryptophan",
"Anthranilic acid",
"5-Hydroxyindole-3-acetic acid",
"N-Acetyl-5-hydroxytryptamine"
)

aas <- c(
"Proline",
"Serine",
"Threonine",
"Valine",
"Alanine", 
"Aspartic acid",
"Glycine",
"Isoleucine",
"Leucine")



lcp4 <- c("#95C65C","#B25D91","#FF3F38")

# joined_df <- join_all(sta_lst_sig[c("Hyperglutamatergic","Hypoglutamatergic")], by="group2", type = "full")
joined_df <- sta_lst_sig
joined_df <- subset(joined_df, variable%in%tryp_mets)
vg <- log_df1[log_df1$variable %in% joined_df$variable, ]
colnames(vg) <- c("rowname","Label","variable","log2 Concentration")
vg$Label <- factor(vg$Label, levels=c("1","10","8"))
maxim_val <- apply(df_norm_lab4[, which(colnames(df_norm_lab4)%in%tryp_mets)], 2, max)
maxim_val <- maxim_val + 0.5
joined_df$y.position <- c(sapply(maxim_val, function(x) rep(x,3)))

four4 <- ggboxplot(vg, x = "Label", y = "log2 Concentration", color = "Label", facet.by = "variable", fill = "Label", alpha=0.4) + geom_point(aes(color = vg$Label)) +
  facet_wrap(~ variable, scales = "free", ncol = 4) +
  stat_pvalue_manual(joined_df, label = "p.adj.signif", y.position = "y.position",hide.ns = TRUE,step.increase = 0.2,bracket.nudge.y = -0.5) +
  labs(title = "Alterations in tryptophan metabolism") +
  theme(
axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(),plot.background = element_rect(fill = "transparent",colour = NA)
) +
   scale_color_manual(values = lcp4, labels = c("Control","Hyperglutamatergic","Hypoglutamatergic")) +
  scale_fill_manual(values = lcp4, labels = c("Control","Hyperglutamatergic","Hypoglutamatergic"))
print(four4) 


ggsave(file=here("paper/Figures/4.png"),plot=four4, dpi = 1000, scale = 0.8, bg="transparent",width = 30, height = 20, units = "cm") 


```

# Alterations in amino acid levels

```{r}
joined_df <- sta_lst_sig
selectedmeta <- setdiff(unique(unlist(sels)),tryp_mets)
joined_df <- subset(joined_df, variable%in%selectedmeta)
vg <- log_df1[log_df1$variable %in% joined_df$variable, ]
colnames(vg) <- c("rowname","Label","variable","log2 Concentration")
vg$Label <- factor(vg$Label, levels=c("1","10","8"))
maxim_val <- apply(df_norm_lab4[, which(colnames(df_norm_lab4)%in%selectedmeta)], 2, max)
maxim_val <- maxim_val 
joined_df$y.position <- c(sapply(maxim_val, function(x) rep(x,3)))

four4 <- ggboxplot(vg, x = "Label", y = "log2 Concentration", color = "Label", facet.by = "variable", fill = "Label", alpha=0.4) + geom_point(aes(color = vg$Label)) +
  facet_wrap(~ variable, scales = "free", ncol = 4) +
  stat_pvalue_manual(joined_df, label = "p.adj.signif", y.position = "y.position",hide.ns = TRUE,step.increase = 0.2) +
  labs(title = "Other alterations") +
  theme(
axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(),plot.background = element_rect(fill = "transparent",colour = NA)
) +
   scale_color_manual(values = lcp4, labels = c("Control","Hyperglutamatergic","Hypoglutamatergic")) +
  scale_fill_manual(values = lcp4, labels = c("Control","Hyperglutamatergic","Hypoglutamatergic"))
print(four4) 


ggsave(file=here("paper/Figures/4B.png"),plot=four4, dpi = 1000, scale = 0.8,width = 30, height = 20, units = "cm") 
```



```{r,fig.height=9,fig.width=10,results='hide'}

# 
# filtered_pathways <- lapply(all_pathways, function(x) if ("map02010" %in% x) x)
# filtered_pathways <- filtered_pathways[!sapply(filtered_pathways, is.null)]
# 
# trnsp <- subset(background, KEGG %in% names(filtered_pathways))$Query
# trnsp <- trnsp[!trnsp%in%amin]
# 
# joined_df <- join_all(sta_lst_sig[c("Hyperglutamatergic","Hypoglutamatergic")], by="group2", type = "full")
# joined_df <- subset(joined_df, variable%in%trnsp)
# maxim_val <- apply(df_norm_lab4[, which(colnames(df_norm_lab4)%in%trnsp)], 2, max)
# vg <- log_df1[log_df1$variable %in% joined_df$variable, ]
# colnames(vg) <- c("rowname","Label","variable","log2 Concentration")
# pg <- ggboxplot(vg, x = "Label", y = "log2 Concentration", fill = "Label", facet.by = "variable") + geom_point() + 
#   facet_wrap(~ variable, scales = "free", ncol = 4) +
#   stat_pvalue_manual(joined_df, label = "p.adj.signif", y = maxim_val-0.01, step.increase = 0.01) +
#   labs(title = "Alterations in other substrates of ABC transporters") +
# theme(
# axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank()
# ) +
#    scale_fill_manual(values = lcp4, labels = c("Control","Hyperglutamatergic","Hypoglutamatergic")) 
# print(pg)

```

```{r}
vip_p <- vip_g
sels_r <- vip_p$Hyperglutamatergic$Metabolite[vip_p$Hyperglutamatergic$p.adj < 0.05 | vip_p$Hyperglutamatergic$VIP > 1]
sels_g <- vip_p$Hypoglutamatergic$Metabolite[vip_p$Hypoglutamatergic$p.adj < 0.05 | vip_p$Hypoglutamatergic$VIP > 1]
# comp_meds <- median_values_t[median_values_t$variable%in%amin,]
# aa_comp <- pivot_wider(data = comp_meds, names_from = Label, values_from = value)
# colnames(aa_comp) <- c("Metabolite","Control","Hypoglutamatergic","Hyperglutamatergic")



```







```{r}
vip_p <- vip_g
vip_p$`Hyperglutamatergic`$diffexpressed <- "None"
vip_p$`Hyperglutamatergic`$diffexpressed[vip_p$`Hyperglutamatergic`$VIP >= 1] <- "MVA"
vip_p$`Hyperglutamatergic`$diffexpressed[vip_p$`Hyperglutamatergic`$p.adj < 0.05] <- "UVA"
vip_p$`Hyperglutamatergic`$diffexpressed[vip_p$`Hyperglutamatergic`$p.adj < 0.05 & vip_p$`Hyperglutamatergic`$VIP > 1] <- "UVA+MVA"
vip_p$`Hyperglutamatergic`$diffexpressed <- factor(vip_p$`Hyperglutamatergic`$diffexpressed, levels = c("UVA+MVA", "MVA", "UVA", "None"))
vip_p$Hypoglutamatergic$diffexpressed <- "None"
vip_p$Hypoglutamatergic$diffexpressed[vip_p$Hypoglutamatergic$VIP >= 1] <- "MVA"
vip_p$Hypoglutamatergic$diffexpressed[vip_p$Hypoglutamatergic$p.adj < 0.05] <- "UVA"
vip_p$Hypoglutamatergic$diffexpressed[vip_p$Hypoglutamatergic$p.adj < 0.05 & vip_p$Hypoglutamatergic$VIP >= 1] <- "UVA+MVA"
vip_p$Hypoglutamatergic$diffexpressed <- factor(vip_p$Hypoglutamatergic$diffexpressed, levels = c("UVA+MVA", "MVA", "UVA", "None"))

```


```{r}


col_op <- c("#B25D91", "#FF3F38","#95C65C")
names(col_op) <- c("Hyperglutamatergic","Hypoglutamatergic", "Control")



two_e <- ggplot(data =  vip_p$Hypoglutamatergic, aes(x = p.adj, y = VIP, label=Metabolite, col=diffexpressed)) +
  geom_vline(xintercept = 0.05, col = "darkgray", linetype = 'dashed') +
  geom_hline(yintercept = 1, col = "darkgray", linetype = 'dashed') + 
  geom_point(size = 3) + 
  # geom_text_repel(aes(label=ifelse(vip_p$Hypoglutamatergic$p.adj < 0.05,as.character(vip_p$Hypoglutamatergic$Metabolite),'')),max.overlaps = 15) +
  scale_color_manual(values = c("#9900F0", "#FF00E4", "#38E54D","grey"), # to set the colors of our variable
                     labels = c("UVA+MVA", "MVA", "UVA", "None"), breaks = levels(vip_p$Hypoglutamatergic$diffexpressed)) +
  theme_light() +
  # coord_cartesian(ylim = c(0, 3.5), xlim = c(0, 3.5)) +
  labs(color = 'Altered', #legend_title, 
       x = "p-value", y = "VIP score", title=NULL) +
  theme(legend.position = "bottom") +
  xlim(c(0,1))

two_f <- ggplot(data=vip_p$`Hyperglutamatergic` ,aes(x = p.adj, y = VIP,  label=Metabolite,col=diffexpressed)) +
  geom_vline(xintercept = 0.05, col = "darkgray", linetype = 'dashed') +
  geom_hline(yintercept = 1, col = "darkgray", linetype = 'dashed') +  
  geom_point(size = 3) + 
  # geom_text_repel(aes(label=ifelse(vip_p$`Hyperglutamatergic`$p.adj < 0.05,as.character(vip_p$`Hyperglutamatergic`$Metabolite),'')),max.overlaps = 15) +
  scale_color_manual(values = c("#9900F0","#FF00E4","#38E54D",  "grey"), # to set the colours of our variable
  labels = c("UVA+MVA", "MVA", "UVA", "None"), breaks = levels(vip_p$`Hyperglutamatergic`$diffexpressed)) +
  theme_light() +
  # coord_cartesian(ylim = c(0, 3.5), xlim = c(0, 3.5)) +
  labs(color = 'Altered', 
       x = "p-value", y = "VIP score", title=NULL) +
  theme(legend.position = "bottom")

ggsave(file=here("paper/Figures/2C.jpeg"),plot=two_e) 
ggsave(file=here("paper/Figures/2D.jpeg"),plot=two_f)

```


```{r}
  vip_g2 <- vip_g$Hyperglutamatergic[order(vip_g$Hyperglutamatergic$VIP, decreasing = TRUE),]
  vip_g2 <- subset(vip_g2, VIP > 1)
  vip_g2$color <- factor(names(vip_g[1]),levels=c("Hyperglutamatergic","Hypoglutamatergic"))
  two_c <- vip_g2 %>%
    filter(!is.na(VIP)) %>%
    arrange(VIP) %>%
    mutate(Metabolite = factor(Metabolite, Metabolite)) %>%
    ggplot(aes(x = Metabolite, y = VIP, size = 1, fill = color)) +
    geom_col() +
    coord_flip() +
    theme_bw(base_size = 12) +
    xlab("") +
    ylab("VIP scores") +
    ylim(0, 4) +
    guides(size = "none") +
    scale_fill_manual(values = col_op[1], labels = "Hyperglutamatergic") +
    theme(legend.title=element_blank(),legend.position = "bottom")
  
  vip_g2 <- vip_g$`Hypoglutamatergic`[order(vip_g$`Hypoglutamatergic`$VIP, decreasing = TRUE),]
  vip_g2 <- subset(vip_g2, VIP > 1)
  vip_g2$color <- factor(names(vip_g[2]),levels=c("Hyperglutamatergic","Hypoglutamatergic"))
  two_d <- vip_g2 %>%
    filter(!is.na(VIP)) %>%
    arrange(VIP) %>%
    mutate(Metabolite = factor(Metabolite, Metabolite)) %>%
    ggplot(aes(x = Metabolite, y = VIP, size = 1, fill = color)) +
    geom_col() +
    coord_flip() +
    theme_bw(base_size = 12) +
    xlab("") +
    ylab("VIP scores") +
    ylim(0, 4) +
    guides(size = "none") +
    scale_fill_manual(values = col_op[2], labels = "Hypoglutamatergic") +
    theme(legend.title=element_blank(),legend.position = "bottom") +
    scale_x_discrete(position = "top")
  

patchvip <- two_c + two_d + plot_layout(guides = "collect") & theme(legend.position = "right", plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA)) 
    
ggsave(file=here("paper/Figures/vip.png"),patchvip,dpi = 1000, scale = 1, bg="transparent",width = 30, height = 12, units = "cm")

```

```{r}
df_metg <- df_norm_lab4 
df_metg <- subset(df_metg,Label%in%c(1,10))
metdat2 <- metdat[order(make.names(metdat$COS.Code)),]
df_metg <- df_metg[order(make.names(rownames(df_metg))),]
rownames(df_metg) <- metdat2[metdat2$COS.Code%in%make.names(rownames(df_metg)),]$ID
df_metg <- data.frame(t(df_metg),check.names = FALSE)
mat_allg <- as.matrix(df_metg)
lab <- data.frame(as.factor(mat_allg[1,]))
lab$as.factor.mat_allg.1.... <- gsub(10,"Hyperglutamatergic     ",lab$as.factor.mat_allg.1....)
lab$as.factor.mat_allg.1.... <- gsub(1,"Control",lab$as.factor.mat_allg.1....)
colnames(lab)<-"Label"
# lab$Label<- factor(lab$Label,levels = c("Control","Hyperglutamatergic","Hypoglutamatergic"),ordered = TRUE)
lab$Label<- factor(lab$Label,levels = c("Control","Hyperglutamatergic     "),ordered = TRUE)
mat_allg <- apply(mat_allg[-1,],c(1,2),as.numeric)

# lc <- c("#95C65C","#B25D91","#FF3F38")
# names(lc) <-  levels(lab$Label)
# lc <- list(Label = lc)
lc <- c("#95C65C","#B25D91")
names(lc) <-  levels(lab$Label)
lc <- list(Label = lc)

p6 <- pheatmap(mat_allg, annotation = lab,clustering_method = "ward.D2",fontsize = 10,show_colnames = TRUE,show_rownames = FALSE, border_color = NA, annotation_colors = lc, fontsize_col = 10, color = colorRampPalette(c("blue","red"))(100), cluster_rows = TRUE,scale = "row",legend = TRUE, annotation_legend = TRUE, cellwidth = 10, cellheight = 2)

p6 <- wrap_elements(as.grob(p6))
```


```{r}
df_metg <- df_norm_lab4 
df_metg <- subset(df_metg,Label%in%c(1,8))
metdat2 <- metdat[order(make.names(metdat$COS.Code)),]
df_metg <- df_metg[order(make.names(rownames(df_metg))),]
rownames(df_metg) <- metdat2[metdat2$COS.Code%in%make.names(rownames(df_metg)),]$ID
df_metg <- data.frame(t(df_metg),check.names = FALSE)
mat_allg <- as.matrix(df_metg)
lab <- data.frame(as.factor(mat_allg[1,]))
lab$as.factor.mat_allg.1.... <- gsub(8,"Hypoglutamatergic",lab$as.factor.mat_allg.1....)
lab$as.factor.mat_allg.1.... <- gsub(1,"Control",lab$as.factor.mat_allg.1....)
colnames(lab)<-"Label"
# lab$Label<- factor(lab$Label,levels = c("Control","Hyperglutamatergic","Hypoglutamatergic"),ordered = TRUE)
lab$Label<- factor(lab$Label,levels = c("Control","Hypoglutamatergic"),ordered = TRUE)
mat_allg <- apply(mat_allg[-1,],c(1,2),as.numeric)

# lc <- c("#95C65C","#B25D91","#FF3F38")
# names(lc) <-  levels(lab$Label)
# lc <- list(Label = lc)

lc <- c("#95C65C","#FF3F38")
names(lc) <-  levels(lab$Label)
lc <- list(Label = lc)

p5 <- pheatmap(mat_allg, annotation = lab,clustering_method = "ward.D2",fontsize = 10,show_colnames = TRUE,show_rownames = FALSE, border_color = NA, annotation_colors = lc, color = colorRampPalette(c("blue","red"))(100), cluster_rows = TRUE,scale = "row",legend = TRUE, annotation_legend = TRUE, cellwidth = 10, cellheight = 2)

p5 <- wrap_elements(as.grob(p5))
```


```{r}


# plot(pd_g$Hypoglutamatergic,typeVc ="x-score",plotSubC="Hypoglutamatergic",parPaletteVc=c("#95C65C","#B25D91"))
# plot(pd_g$`Hyperglutamatergic`,typeVc ="x-score",plotSubC="Hypoglutamatergic",parPaletteVc=c("#95C65C","#FF3F38"))
rt <- gsub("10", "Hyperglutamatergic", subset(df_norm_lab4,Label%in%c(1,10))$Label)
rt <- gsub("1", "Control", rt)
rt <- factor(rt,levels = c("Control","Hyperglutamatergic","Hypoglutamatergic"))
gr <- gsub("8", "Hypoglutamatergic", subset(df_norm_lab4,Label%in%c(1,8))$Label)
gr <- gsub("1", "Control", gr)
gr <- factor(gr,levels = c("Control","Hyperglutamatergic","Hypoglutamatergic"))

opdf <- data.frame("lab"=rt,"x"=pd_g$Hyperglutamatergic@scoreMN,"y"=pd_g$Hyperglutamatergic@orthoScoreMN)
opdf2 <- data.frame("lab"=gr,"x"=pd_g$`Hypoglutamatergic`@scoreMN,"y"=pd_g$`Hypoglutamatergic`@orthoScoreMN)

# par(mfrow=c(1,2))
two_a <- opdf %>%
  ggplot(aes(x=p1,y=o1,color=lab)) +
  geom_point(size=3) +
  stat_ellipse(geom = "polygon",
               aes(fill = lab), 
               alpha = 0.25,
               level = 0.8) +
  coord_fixed(xlim = c(-10,10),ylim=c(-5,5)) +
  theme_bw(base_size = 15) +
  # ggtitle(paste("OPLS-DA", "SCORE PLOT ",collapse="\n")) +
  scale_color_manual(values=c("#95C65C","#B25D91","#FF3F38"),labels=c("Control","Hyperglutamatergic","Hypoglutamatergic"),drop = FALSE) +
  scale_fill_manual(values=c("#95C65C","#B25D91","#FF3F38"),labels=c("Control","Hyperglutamatergic","Hypoglutamatergic"),drop = FALSE) +
  theme(legend.title = element_blank()) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept = 0)

two_b <- opdf2 %>%
  ggplot(aes(x=p1,y=o1,color=lab)) +
  geom_point(size=3) +
  stat_ellipse(geom = "polygon",
               aes(fill = lab), 
               alpha = 0.25,
               level = 0.8) +
  coord_fixed(xlim = c(-10,10),ylim=c(-5,5)) +
  theme_bw(base_size = 15) +
  # ggtitle(paste("OPLS-DA", "SCORE PLOT ",collapse="\n")) +
  scale_color_manual(values=c("#95C65C","#B25D91","#FF3F38"),drop = FALSE) +
  scale_fill_manual(values=c("#95C65C","#B25D91","#FF3F38"),drop = FALSE) +
  theme(legend.title = element_blank()) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept = 0) 
  

patchop <- two_a + two_b + plot_annotation("OPLS-DA SCORE PLOTS")  + plot_layout(guides = "collect") & theme(legend.position = 'top')  
 
ggsave(file=here("paper/Figures/opls.jpeg"),plot=patchop, width = 300,
       height = 200,
       units = "mm", dpi = 300, scale = 0.8)

# ggsave(file=here("paper/Figures/2A.jpeg"),plot=two_a) 
# ggsave(file=here("paper/Figures/2B.jpeg"),plot=two_b) 

# op <- wrap_elements(two_a + two_b + plot_annotation(tag_levels = 'A') + plot_layout(guides = "keep",widths = c(1,1),heights = c(1,1)) & theme(legend.position = 'bottom'))

# two_g <- wrap_elements(as.grob(two_g))
# two_h <- wrap_elements(as.grob(two_h))
# 
# 
# (op/(two_c|two_d)/(two_e|two_f)/(two_g|two_h)+plot_annotation(tag_levels = 'A'))

fig_2 <- (p6 + plot_spacer() + p5  + plot_layout(widths = c(1, 1,1))) / (two_a + two_b + plot_layout(guides = "collect") & theme(legend.position="bottom")) /  (two_c + two_d + plot_layout(guides = "collect") & theme(legend.position="bottom"))+ plot_layout(heights = c(2,1,1),widths = c(3,1,1)) + plot_annotation(tag_levels = 'A') & theme(plot.tag = element_text(size = 15, hjust = 0, vjust = 0)) 
fig_2
ggsave(file=here("paper/Figures/fig_2.png"),fig_2,dpi = 1000, scale = 1,width = 30, height = 30, units = "cm")


# (two_c + two_d + plot_layout(guides = "collect")
```

```{r}

# df_norm_lab6 <- df_norm_lab4[,c("rowname","Label",aas)]
# log_df_aa <- melt(df_norm_lab6[,-1],id.vars = "Label")
# man <- manova(cbind(Phenylalanine, Proline, Serine, Threonine, Tryptophan, Valine, Alanine, `Aspartic acid`, Glutamine, `Glutamic acid`, Glycine, Isoleucine, Leucine)~Label,df_norm_lab6)
# 
# summary(man)
# summary.aov(man)
# 
# an <- aov(value~Label*variable,log_df_aa)
# summary(an)
# summary.aov(an)
```


```{r}
library(psych)

metdat_char <- read.csv("C:/Users/killescas/OneDrive - San Juan de Dios/Desktop/TESIS/LCR/metabolomica/meds_5h.csv",sep=";")
metdat_char <- metdat_char[complete.cases(metdat_char),]
metdat_char <- metdat_char %>%
  column_to_rownames(var="ID")
contab <- table(metdat_char)
phi(contab,digits=3)


```



